cnd/src/asset/bitcoin.rs

```diff
@@ -0,0 +1,55 @@
+use bitcoin::{
+    util::amount::{Denomination, ParseAmountError},
+    Amount,
+};
+use std::fmt;
+
+#[derive(Debug, Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub struct Bitcoin(Amount);
+
+impl Bitcoin {
+    pub fn from_sat(sat: u64) -> Bitcoin {
+        Bitcoin(Amount::from_sat(sat))
+    }
+
+    pub fn as_sat(self) -> u64 {
+        Amount::as_sat(self.0)
+    }
+
+    pub fn from_btc(btc: f64) -> Result<Bitcoin, ParseAmountError> {
```

Comment:
We actually only ever use this method in tests, so you could just remove it and change the tests.

Using floats for money is dangerous. If we don't have to provide this method, that would be very good :D